README.md

```diff
@@ -186,10 +186,10 @@ Minikube tools to be installed and available on your PATH.
 
    * `vs-kubernetes` - Parent for Kubernetes-related extension settings
        * `vs-kubernetes.namespace` - The namespace to use for all commands
-       * `vs-kubernetes.kubectl-path` - File path to the kubectl binary. Note this is the binary file itself, not just the directory containing the file. On Windows, this must contain the `.exe` extension.
+       * `vscode-kubernetes.kubectl-path` - File path to the kubectl binary. Note this is the binary file itself, not just the directory containing the file. On Windows, this must contain the `.exe` extension.
```

Comment:
As you allude to in your comment, the new paths are not nested.  The non-nesting is why they exist - the trust model only allows us to identify _top-level_ config settings as machine-scope (and therefore not overridable by workspaces).  With the nested settings, we could only have set a policy on the _whole_ `vs-kubernetes` wrapper object.  Hence the top-level settings.  These are now the preferred ones; the old nested ones exist for backward compatibility.

In addition, as you have spotted, the names are slightly different.  The old keys use `vs-kubernetes` for compatibility all the way back to Brendan's original extension.  The new ones use `vscode-kubernetes` because that's closer to what the extension is now called.

So the story should be:

* Document the top-level settings
* Remove the documentation for the nested settings
* Use the top-level settings if present and fall back to the nested ones if not (this should already be handled by `components/config/config.ts/getToolPath`)

You can see the structure in `package.json` but you may have to do a lot of collapsing because there are a lot of nested settings between the big object and the top-level settings!

Reply:
Aha, thank you so much for the background ❤️☕️🙏, yes I was and only after searching found out the `deprecated` bits, sound good, this seems like careful removal work, I will try and continue the path mentioned above and make the changes. Thanks heaps.